src/OpenLoco/src/Map/TileManager.cpp

```diff
@@ -1267,6 +1268,86 @@ namespace OpenLoco::World::TileManager
         return totalCost;
     }
 
+    // 0x004C4C28
+    uint32_t adjustWaterHeight(World::Pos2 pos, SmallZ targetHeight, std::set<World::Pos3, LessThanPos3>& removedBuildings, uint8_t flags)
+    {
+        GameCommands::setExpenditureType(ExpenditureType::Construction);
+        GameCommands::setPosition(World::Pos3(pos.x + World::kTileSize / 2, pos.y + World::kTileSize / 2, targetHeight * kMicroToSmallZStep));
+
+        if (targetHeight < 4)
+        {
+            GameCommands::setErrorText(StringIds::error_too_low);
+            return GameCommands::FAILURE;
+        }
+
+        if (targetHeight > 116)
+        {
+            GameCommands::setErrorText(StringIds::error_too_high);
+            return GameCommands::FAILURE;
+        }
+
+        currency32_t totalCost = 0;
+
+        if (flags & GameCommands::Flags::apply)
+        {
+            removeSurfaceIndustry(pos);
+
+            if (!isEditorMode())
+            {
+                setTerrainStyleAsCleared(pos);
+            }
+
+            auto clearHeight = getHeight(pos).landHeight;
+            removeAllWallsOnTile(toTileSpace(pos), clearHeight / 4);
+        }
+
+        auto tile = get(pos);
+        auto* surface = tile.surface();
+
+        auto referenceZ = surface->baseZ();
+        auto water = surface->water();
+        if (water > 0)
+        {
+            referenceZ = surface->waterHeight();
+        }
+
+        if (targetHeight < referenceZ)
+        {
+            targetHeight = referenceZ;
+        }
+
+        // Bind our local vars to the tile clear function
+        auto clearFunc = [pos, &removedBuildings, flags, &totalCost](TileElement& el) {
+            return TileClearance::clearWithDefaultCollision(el, pos, removedBuildings, flags, totalCost);
+        };
+
+        QuarterTile qt(0xF, 0);
+        auto testHeight = TileManager::getSurfaceCornerHeight(surface);
+        if (!TileClearance::applyClearAtAllHeights(pos, testHeight / 4, testHeight / 4, qt, clearFunc))
+        {
+            return GameCommands::FAILURE;
+        }
+
+        if (surface->isFlag6())
+        {
+            GameCommands::setErrorText(StringIds::water_channel_currently_needed_by_ships);
+            return GameCommands::FAILURE;
+        }
+
+        auto* waterObj = ObjectManager::get<WaterObject>();
+        totalCost += Economy::getInflationAdjustedCost(waterObj->costFactor, waterObj->costIndex, 10);
+
+        if (flags & GameCommands::Flags::apply && surface->baseZ() < targetHeight)
```

Comment:
nah you can't do this as you need to always set the following and you need to set water to 0